File: app/src/main/java/com/letianpai/robot/control/system/LetianpaiLightUtil.java
package com.letianpai.robot.control.system;

import android.content.Context;
import android.provider.Settings;
import android.util.Log;
import android.view.WindowManager;

/**
 * @author liujunbin
 */
public class LetianpaiLightUtil {

    public static int getLightMode(Context context) {
        // 获取屏幕亮度模式
        int mode = 0;
        try {
            mode = Settings.System.getInt(context.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS_MODE);
        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
        }
        return mode;
    }

    /**
     * 获取当前屏幕亮度值
     *
     * @param context
     * @return
     */
    public static int getCurrentBrightness(Context context) {
        int currentBrightness = 0;
        try {
            currentBrightness = Settings.System.getInt(context.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS);
        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
        }
        return currentBrightness;
    }

    /**
     * 设置屏幕亮度模式为手动模式
     * @param context
     */
    public static void setScreenBrightnessToManualMode(Context context) {
        Settings.System.putInt(context.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS_MODE, Settings.System.SCREEN_BRIGHTNESS_MODE_MANUAL);
    }

    /**
     * 设置屏幕亮度值为
     * @param context
     */
    public static void setScreenBrightness(Context context,int value) {
        //  50（取值范围为 0~255）
        Settings.System.putInt(context.getContentResolver(),Settings.System.SCREEN_BRIGHTNESS,value);
    }

    /**
     * 设置屏幕亮度值为
     * @param context
     */
    public static void setScreensBrightness(Context context,int value) {
        //  50（取值范围为 0~255）
        Settings.System.putInt(context.getContentResolver(),Settings.System.SCREEN_BRIGHTNESS,value);
    }

    /**
     * 设置屏幕亮度值为
     * @param context
     */
    public static void setScreensBrightness1(Context context,int value) {
        //  50（取值范围为 0~255）
        Settings.System.putInt(context.getContentResolver(),Settings.System.SCREEN_BRIGHTNESS,value);
    }

}

File: app/src/main/java/com/letianpai/robot/control/system/LetianpaiLightUtil.kt
package com.letianpai.robot.control.system

import android.content.Context
import android.provider.Settings
import android.provider.Settings.SettingNotFoundException

/**
 * @author liujunbin
 */
object LetianpaiLightUtil {
    fun getLightMode(context: Context): Int {
        // Getting the screen brightness mode
        var mode: Int = 0
        try {
            mode = Settings.System.getInt(
                context.contentResolver,
                Settings.System.SCREEN_BRIGHTNESS_MODE
            )
        } catch (e: SettingNotFoundException) {
            e.printStackTrace()
        }
        return mode
    }

    /**
     * Get the current screen brightness value
     *
     * @param context
     * @return
     */
    fun getCurrentBrightness(context: Context): Int {
        var currentBrightness: Int = 0
        try {
            currentBrightness = Settings.System.getInt(
                context.contentResolver,
                Settings.System.SCREEN_BRIGHTNESS
            )
        } catch (e: SettingNotFoundException) {
            e.printStackTrace()
        }
        return currentBrightness
    }

    /**
     * Setting the screen brightness mode to manual
     * @param context
     */
    fun setScreenBrightnessToManualMode(context: Context) {
        Settings.System.putInt(
            context.contentResolver,
            Settings.System.SCREEN_BRIGHTNESS_MODE,
            Settings.System.SCREEN_BRIGHTNESS_MODE_MANUAL
        )
    }

    /**
     * Set the screen brightness value to
     * @param context
     */
    fun setScreenBrightness(context: Context, value: Int) {
        //  50（The range of values is 0~255）
        Settings.System.putInt(
            context.contentResolver,
            Settings.System.SCREEN_BRIGHTNESS,
            value
        )
    }

    /**
     * Set the screen brightness value to
     * @param context
     */
    fun setScreensBrightness(context: Context, value: Int) {
        // 50（The range of values is 0~255）
        Settings.System.putInt(
            context.contentResolver,
            Settings.System.SCREEN_BRIGHTNESS,
            value
        )
    }

    /**
     * Set the screen brightness value to
     * @param context
     */
    fun setScreensBrightness1(context: Context, value: Int) {
        // 50（The range of values is 0~255））
        Settings.System.putInt(
            context.contentResolver,
            Settings.System.SCREEN_BRIGHTNESS,
            value
        )
    }
}
